Reduce UI Options on Main Graphic Page

nlin21 opened the pull request:
# Description

These changes reduce the amount of UI options on the left side menu, moving the remaining options into a modal popup. This modal popup shows up when the 'More options' button is pressed, and its contents are customized for each chart type.

<img width="232" alt="image" src="https://github.com/user-attachments/assets/96ca2813-b708-495c-a1fa-2f1db4d9ce95">

<img width="1438" alt="image" src="https://github.com/user-attachments/assets/ae1878ae-db8b-4d39-8702-e1ad92ba4152">

Fixes #900

## Type of change
- [ ] Note merging this changes the database configuration.
- [x] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement](https://openenergydashboard.github.io/developer/cla.html) and each author is listed in the Description section.

## Limitations

There are a few issues I came across:
1. Extra colon, "Area Unit::"
<img width="140" alt="image" src="https://github.com/user-attachments/assets/05f29d24-da3f-4423-bdeb-515a64936e6d">

2. Vertical spacing between components are not consistent, I am not sure why this is.

nlin21 committed:
Setup for use with MoreOptionsComponent

nlin21 committed:
Move MoreOptionsComponent to appropriate directory, set up test button

nlin21 committed:
Assign Modal popup to button click, move over necessary imports from UIOptionsComponent

nlin21 committed:
Add custom options for each chart type

huss commented on src/client/app/components/MoreOptionsComponent.tsx:
```diff
@@ -0,0 +1,82 @@
+			{
+				<div>
+					<Button color='secondary' outline onClick={handleShow}>
+						More Options
```
The text needs to be internationalized.

huss commented on src/client/app/components/MoreOptionsComponent.tsx:
```diff
@@ -0,0 +1,82 @@
+						More Options
+					</Button>
+					<Modal isOpen={showModal} toggle={handleClose} size='md'>
+						<ModalHeader>More Options for '{chartToRender}' Graph Type:</ModalHeader>
```
I have mixed feelings about putting the graph type on the title. It makes it longer and OED does not tend to do this in other places. I do see it is more precise. Overall, my inclination is to just use "More Options" but I'm open to other thoughts.

This needs to be internationalized.

huss committed:
fix space below area unit

huss committed:
rearrange map item location

Moves map choice above interval and places tooltip with
title of map as with other pages

huss committed:
add space above more button

This is not a perfect fix for the styling issues but does make
sure it does not abut the item above.

huss reviewed (changes requested):
Thanks to @nlin21 for another contribution. This works well and does what was desired. I have a couple of minor comments to consider.

Also, I pushed a few commits that addressed items that are mostly outside the original scope of the PR. However, they address minor UI option issues that were outstanding and this seemed a reasonable and easy place to do them. They are described in the commit msg. The only one directly related to this PR is to add space above the more option button.

nlin21 committed:
Internationalize 'more options' text

huss commented:
Thanks to @nlin21 for the updates. Everything looks good.
